### ISSUE TITLE
Write more client DNS tests

### ISSUE THREAD

**Issue by jamilbk** [CloudFire-LLC/cloudfire#3342]
When #1762 has a working Split DNS implementation, we should make sure those core components are tested:

- [x] DNS integration tests in CI
  - [x] Add a `test.httpbin.docker.local` service in `docker-compose.yml`
  - [x] Add `?.httpbin.docker.local` and `*.docker.local` as Resources in the test env seeds.exs
  - [x] Test that client can ping them
  - [x] Test that they returns CGNAT IP
  - [x] Test that lookups for e.g. github.com still work and do not return with CGNAT IP
- [ ] Unit tests
  - [ ] Test more combinations of DNS lookups using mocks in Rust: Single name resolving to multiple IPv4 and IPv6 addresses. Single IPv4/6 being resolved to by multiple different subdomains and TLDs.

**Comment by thomaseizinger** [comment]
Here are the integration tests for `firezone-connection`: https://github.com/firezone/firezone/tree/main/rust/connection-tests

The main difference to the main `docker-compose.yml` is that this one is setting up a network topology where both clients are behind a NAT.

To test DNS, we could add a another container (a resource) next to the gateway (in the test called `listener`) that the client will try to reach via a DNS name.

**Comment by jamilbk** [comment]
Should be implemented as part of #1762

**Comment by ReactorScram** [comment]
Maybe this should be in the Windows smoke test too?

**Comment by jamilbk** [comment]
- will need to remove hardcoding the sentinel DNS
- blocked on #3323

**Comment by jamilbk** [comment]
> Maybe this should be in the Windows smoke test too?

The issue with testing the other clients is that `docker` isn't available on those GH runners :-(

There are [service containers](https://docs.github.com/en/actions/using-containerized-services/about-service-containers) but those must be specified on the job and are Linux-only, so there's not a good way to test cross-host integration tests

**Comment by thomaseizinger** [comment]
I am hoping that we can more easily test this once the tunnel is sans-io as it should be possible to run a client and a gateway tunnel next to each other in-memory without having them any actual IO.

**Comment by ReactorScram** [comment]
What's the next step for this one?

I think we have one integration test that uses Docker Compose to reach `httpbin.test` over DNS.

**Comment by jamilbk** [comment]
I think we can save further DNS tests for another PR, marking this as done (feel free to reopen otherwise)
